### ISSUE TITLE
Debug output from Chromecast

### ISSUE THREAD

**Issue by JonasVautherin** [googlecast/CastReceiver#47]
Is there a way to get a debug output from the Chromecast? I can successfully cast some videos, but some don't work, and I am having a hard time understanding why (it's an HLS setup).

Would be great if I could see why the Chromecast is not happy, like "Some CORS headers are missing" or "invalid video format" or something, instead of blind guessing.

Is there a way to do that?

**Comment by joeyparrish** [comment]
Yes, but there are a few steps you have to take to get that ability:

1. In the [Cast SDK Developer Console](https://cast.google.com/publish/), register a receiver app (I assume you've done this at least)
2. In the console, register your Chromecast device by its serial number (this is easy to get wrong, since it might contain 1, I, and/or l)
3. Reboot the Chromecast device so that it gets its new config

When you start any receiver app registered to you, devices also registered to you will allow you to open a remote dev console through Chrome and inspect and/or debug the receiver app.  To do this, go to `chrome://inspect` and click the "inspect" link under your device in the list.

Here's a YouTube video that shows someone opening the remote dev console on a Chromecast: https://www.youtube.com/watch?v=EL2Eomyce9o

If you don't see "inspect", here are some things to check:

1. Are you on the same network with your Chromecast?
2. Did you enter the correct serial number?  (It's very easy to make mistakes on the serial number.)
3. Wait an hour and reboot the Chromecast device again.  (Maybe the registration hadn't propagated and been sent to the device yet.)

If you want help getting the device serial number right, there is a helpful trick in the registration dialog.  A small speaker icon can be used to launch a receiver app that reads the device serial number out loud to you.  Here's a screenshot of that icon in the registration dialog:

![7k9KmcbsEdxfo5L](https://user-images.githubusercontent.com/6843156/204330802-74f20a60-ae00-45a3-ac54-1e5bfefa0caf.png)

I hope this helps!

**Comment by JonasVautherin** [comment]
Thanks a lot for the quick answers! 

> I assume you've done this at least

I have actually tried a ton of stuff, and I don't think there are many Google/DuckDuckGo results I haven't read in the last few days :see_no_evil:. This included, when running https://developers.google.com/cast/codelabs/cast-videos-chrome#2.

> this is easy to get wrong, since it might contain 1, I, and/or l

Yep, for me it was O and 0, and the fact that nothing on the device says that it is the S/N, so I just registered all the numbers I found there, until I found the chrome trick that read "zero" for my "O" :sweat_smile:. But that's fine, only cost an hour.

I was hoping that I could get some log output with `cast.framework.CastReceiverContext.getInstance().setLoggerLevel(cast.framework.LoggerLevel.DEBUG);`, as described [here](https://developers.google.com/cast/docs/debugging/remote_debugger), but I never managed to get anywhere in the console (either the functions do not exist, or they return `undefined`).

By modifying https://developers.google.com/cast/codelabs/cast-videos-chrome#2 and pointing the video to my local HTTP server (serving an HLS video), I managed to see it try to load my video and fail the same way as when I use `pychromecast` (i.e. it starts loading and suddenly stops). I'm not seeing any kind of useful debug output in what I think is called the "Remote Debugger" (where the image on the TV is mirrored in Google-Chrome).

I guess I won't get much further with this crazy idea of getting some debug output from my Chromecast device, after all :confused:.
